README.md

```diff
@@ -1,52 +1,205 @@
+  name: "default"
+spec:
+  serviceNetwork: "172.30.0.0/16"
+  clusterNetworks:
+    - cidr: "10.128.0.0/14"
+      hostSubnetLength: 9
+  defaultNetwork:
+    type: OpenShiftSDN
 ```
 
-## Running
+## Configuring IP address pools
+Users must supply at least two address pools - one for pods, and one for services. These are the ClusterNetworks and ServiceNetwork parameter. Some network plugins, such as OpenShiftSDN, support multiple ClusterNetworks.
 
-There are some premade Kubernetes manifests to run a development build. After setting the image URL to something sane in the daemonset, do:
-
-```
-oc create -f manifests/
-oc create -f sample-config.yaml
-```
+Currently, changing the address pools once set is not supported. In the future, some network providers may support expanding the address pools.
 
-Then you can watch the daemonset with `kubectl -n openshift-cluster-network-operator logs <PODID>`.
-
-## Example configuration
+Each ClusterNetwork entry has an additional required parameter, `hostSubnetLength`, that specifies the address size to assign to assign to each individual node. Note that this is currently *reverse* from the usual CIDR notation - a hostSubnetLength of 9 means that the node will be assigned a /23.
 
+Example
 ```yaml
-apiVersion: "networkoperator.openshift.io/v1"
-kind: "NetworkConfig"
-metadata:
-  name: "default"
-  namespace: cluster-network-operator
 spec:
   serviceNetwork: "172.30.0.0/16"
   clusterNetworks:
     - cidr: "10.128.0.0/14"
       hostSubnetLength: 9
+    - cidr: "192.168.0.0/18"
+      hostSubnetLength: 9
+```
+
+## Configuring the default network provider
+Users must select a default network provider. This cannot be changed. Different network providers have additional provider-specific settings.
+
+Currently, the only supported value for network Type is `OpenShiftSDN`.
+
+### Configuring OpenShiftSDN
+OpenShiftSDN supports the following configuration options, all of which are optional:
+* `mode`: one of "Subnet" "Multitenant", or "NetworkPolicy". Configures the [isolation mode](https://docs.openshift.com/container-platform/3.11/architecture/networking/sdn.html#overview) for OpenShift SDN. The default is "NetworkPolicy".
+* `vxlanPort`: The port to use for the VXLAN overlay. The default is 4789
+* `MTU`: The MTU to use for the VXLAN overlay. The default is the MTU of the node, minus 50 bytes for overhead. However, this value will never change.
+* `useExternalOpenvswitch`: boolean. If the nodes are already running openvswitch, and OpenShiftSDN should not install its own, set this to true.
+
+Example:
+```yaml
+spec:
   defaultNetwork:
     type: OpenShiftSDN
     openshiftSDNConfig:
       mode: NetworkPolicy
+      vxlanPort: 4789
+      mtu: 1450
+      useExternalOpenvswitch: false
+```
+
+## Configuring kube-proxy
+Users may customize the kube-proxy configuration. None of these settings are required:
+
+* `iptablesSyncPeriod`: The interval between periodic iptables refreshes. Default: 30 seconds.
```

Comment:
What is the valid range? How does changing this impact the behavior of the cluster? Any special cases?